### PR TITLE
3987 - DataExport: GrowingStockComposition -> always export most recent year

### LINKED ISSUE #3987: DataExport: GrowingStockComposition -> always export most recent year
Image

in data export , always export most recent year as first column after country and remove it from variables selection

### PR DESCRIPTION
**Pull request by yaguzmang**
image

After year format:
image

### CODE REVIEW

File: src/test/migrations/steps/20241002151455-step-add-cellsExportAlways-to-growingStockComposition2025.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      from ${schemaAssessment}."table" t2
+      left join ${schemaAssessment}.table_section ts on t2.table_section_id = ts.id
+      left join ${schemaAssessment}.section s on s.id = ts.section_id
+      where s.props ->> 'name' = 'growingStockComposition'
+      and t2.props ->> 'name' = 'growingStockComposition2025'
+    ) as src
+    where t.id = src.id;
+  `)
+
+  await AssessmentController.generateMetadataCache({ assessment }, client)
```
Comment (minotogna) [inline src/test/migrations/steps/20241002151455-step-add-cellsExportAlways-to-growingStockComposition2025.ts]: I think we can have only 1 migration step ?

File: src/client/pages/DataExport/ResultsTable/Title/Title.tsx [hunk 2]
```diff
@@ -4,8 +4,8 @@ import { useParams } from 'react-router-dom'
 
 import { Objects } from 'utils/objects'
 
-import { Cols, Row } from 'meta/assessment'
-import { Unit, UnitFactors } from 'meta/dataExport'
+import { Cols, Row, Unit } from 'meta/assessment'
```
Comment (minotogna) [inline src/client/pages/DataExport/ResultsTable/Title/Title.tsx:7]: Is meta/dataExport/Unit still needed ?

**Comment by sorja** [comment]
Should the year format be `2 021` => `2021`

Also
`Most recent year:` => `Most recent year`

File: src/client/pages/DataExport/ResultsTable/Title/Title.tsx (hunk 2 above)
Comment (yaguzmang) [reply src/client/pages/DataExport/ResultsTable/Title/Title.tsx:7]: It was used in one other place, but now I replaced it with `assessment Unit` and remove it :+1:

Comment (minotogna) [reply src/client/pages/DataExport/ResultsTable/Title/Title.tsx:7]: Nice 👍

File: src/meta/assessment/table.ts [hunk 3]
```diff
@@ -27,12 +27,28 @@ export enum TableNames {
...
+
+export type TableCellNames = Record<CycleUuid, Array<TableCell>>
+
 // array of column names indexed by cycle uuid
 export type TableColumnNames = Record<CycleUuid, Array<ColName>>
 
 export type TableName = string
 
 export interface TableProps {
+  cellsExportAlways?: TableCellNames
```
Comment (minotogna) [inline src/meta/assessment/table.ts:51]: Can u please check the table clone method as well and table repository adapter ?

File: src/test/migrations/steps/20241003084429-step-add-cellsExportAlways-to-growingStockComposition2025-and-excludeFromDataExport-to-row.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    set props = props
+        || jsonb_build_object(
+            'cellsExportAlways',
+            jsonb_build_object(
+                '${cycle.uuid}',
+                '[{
+                    "columnName": "mostRecentYear",
+                    "variableName": "mostRecentYear",
+                    "unit": null,
+                    "format": "year"
```
Comment (minotogna) [inline src/test/migrations/steps/20241003084429-step-add-cellsExportAlways-to-growingStockComposition2025-and-excludeFromDataExport-to-row.ts]: Should we instead just have format: original ? ( to its original value ?)

File: src/meta/assessment/table.ts (hunk 3 above)
Comment (yaguzmang) [reply src/meta/assessment/table.ts:51]: Good catch 👍 thanks
Added cellsExportAlways to clone props. And the table adapter doesn't seem to need a change for the props:
image

Comment (minotogna) [reply src/meta/assessment/table.ts:51]: Nice 👍

**Review by minotogna — APPROVED** [review]

**Pull request merged**